File: backend/app/schemas/task.py
from datetime import datetime
from pydantic import BaseModel
from .category import CategorySchema

class TaskCreate(BaseModel):
    texto_tarea: str
    fecha_creacion: datetime
    fecha_tentativa_finalizacion: datetime | None
    estado: str
    id_Usuario: int
    id_Categoria: int

class TaskUpdate(BaseModel):
    texto_tarea: str | None
    estado: str | None

class TaskSchema(BaseModel):
    id: int
    texto_tarea: str
    fecha_creacion: datetime
    fecha_tentativa_finalizacion: datetime | None
    estado: str
    id_Usuario: int
    id_Categoria: int
    categoria: CategorySchema

    class Config:
        orm_mode = True  # To work seamlessly with SQLAlchemy models

File: frontend/src/types/task.ts
import type { Category } from "./category"

export type Task = {
        id: number,
        texto_tarea: string,
        fecha_creacion: string,
        fecha_tentativa_finalizacion: string,
        estado: string,
        id_Usuario: number,
        id_Categoria: number
        categoria: Category
}
